src/hooks/api/getters/useStrategy/useStrategy.ts

```diff
@@ -0,0 +1,47 @@
+import useSWR, { mutate, SWRConfiguration } from 'swr';
+import { useEffect, useState } from 'react';
+import { formatApiPath } from '../../../../utils/format-path';
+import { IStrategy } from '../../../../interfaces/strategy';
+import handleErrorResponses from '../httpErrorResponseHandler';
+
+const useStrategy = (strategyName: string, options: SWRConfiguration = {}) => {
+    const STRATEGY_CACHE_KEY = `api/admin/strategies/${strategyName}`;
+    const path = formatApiPath(STRATEGY_CACHE_KEY);
+
+    const fetcher = () => {
+        return fetch(path, {
+            method: 'GET',
+            credentials: 'include',
+        })
+            .then(handleErrorResponses(`${strategyName} strategy`))
+            .then(res => res.json());
+    };
+
+    const { data, error } = useSWR<{ strategy: IStrategy }>(
+        STRATEGY_CACHE_KEY,
+        fetcher,
+        options
+    );
+    const [loading, setLoading] = useState(!error && !data);
```

Comment:
Don't need state for `loading`, could return `loading: !error && !data` instead.